bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/WLedHandler.java

```diff
@@ -0,0 +1,420 @@
+            }
+        } catch (TimeoutException e) {
+            errorReason = "TimeoutException: WLED was not reachable on your network";
+        } catch (ExecutionException e) {
+            errorReason = String.format("ExecutionException: %s", e.getMessage());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            errorReason = String.format("InterruptedException: %s", e.getMessage());
+        }
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorReason);
+    }
+
+    private HSBType parseToHSBType(String message, String element) {
+        int startIndex = message.indexOf(element);
+        if (startIndex == -1) {
+            return new HSBType();
+        }
+        int endIndex = message.indexOf("<", startIndex + element.length());
+        int r = 0, g = 0, b = 0;
+        try {
+            r = Integer.parseInt(message.substring(startIndex + element.length(), endIndex));
+            // look for second element
+            startIndex = message.indexOf(element, endIndex);
+            if (startIndex == -1) {
+                return new HSBType();
+            }
+            endIndex = message.indexOf("<", startIndex + element.length());
+            g = Integer.parseInt(message.substring(startIndex + element.length(), endIndex));
+            // look for third element called <cl>
+            startIndex = message.indexOf(element, endIndex);
+            if (startIndex == -1) {
+                return new HSBType();
+            }
+            endIndex = message.indexOf("<", startIndex + element.length());
+            b = Integer.parseInt(message.substring(startIndex + element.length(), endIndex));
+        } catch (NumberFormatException e) {
+            logger.warn("NumberFormatException when parsing the WLED color fields.");
+        }
+        return HSBType.fromRGB(r, g, b);
+    }
+
+    private void parseColours(String message) {
+        primaryColor = parseToHSBType(message, "<cl>");
+        updateState(CHANNEL_PRIMARY_COLOR, primaryColor);
+        secondaryColor = parseToHSBType(message, "<cs>");
+        updateState(CHANNEL_SECONDARY_COLOR, secondaryColor);
+        try {
+            primaryWhite = new BigDecimal(WLedHelper.getValue(message, "<wv>", "<"));
+            if (primaryWhite.intValue() > -1) {
+                hasWhite = true;
+                updateState(CHANNEL_PRIMARY_WHITE,
+                        new PercentType(primaryWhite.divide(BIG_DECIMAL_2_55, RoundingMode.HALF_UP)));
+                secondaryWhite = new BigDecimal(WLedHelper.getValue(message, "<ws>", "<"));
+                updateState(CHANNEL_SECONDARY_WHITE,
+                        new PercentType(secondaryWhite.divide(BIG_DECIMAL_2_55, RoundingMode.HALF_UP)));
+            }
+        } catch (NumberFormatException e) {
+            logger.warn("NumberFormatException when parsing the WLED colour and white fields.");
```

Comment:
At the very least you should log the payload that failed to parse. That way a user would have a better idea of what went wrong.

Reply:
DONE